src/java/org/jivesoftware/openfire/plugin/SearchPlugin.java

```diff
@@ -732,23 +732,27 @@ private IQ replyDataFormResult(Collection<User> users, IQ packet) {
         searchResults.addReportedField("jid", "JID", FormField.Type.jid_single);
 
         for (final String fieldName : getFilteredSearchFields()) {
+            String fieldLabel = LocaleUtils.getLocalizedString("advance.user.search." + fieldName.toLowerCase(), "search");
             searchResults.addReportedField(fieldName,
-                    LocaleUtils.getLocalizedString("advance.user.search." + fieldName.toLowerCase(), "search"), FormField.Type.text_single);
+                fieldLabel, FormField.Type.text_single);
         }
 
         for (final User user : users) {
             final String username = JID.unescapeNode(user.getUsername());
+            String jid = username + "@" + serverName;
```

Comment:
To further improve this: The `jid` should be constructed using the `JID` class, as that will apply appropriate string-prepping.

You can use something like `XMPPServer.getInstance().createJID(username, null)` and then use JID's `toString()`.